confmap/internal/featuregate.go

```diff
@@ -0,0 +1,14 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package internal // import "go.opentelemetry.io/collector/confmap/internal"
+
+import "go.opentelemetry.io/collector/featuregate"
+
+const StrictlyTypedInputID = "confmap.strictlyTypedInput"
+
+var StrictlyTypedInputGate = featuregate.GlobalRegistry().MustRegister(StrictlyTypedInputID,
```

Comment:
nit: curious why this needs to be in internal? There may be contrib packages that want to test against this feature gate

Reply:
No strict need for this to be internal, but I would like to wait until someone asks for it to make it public